src/test/isolation2/expected/modify_table_data_corrupt_optimizer.out

```diff
@@ -163,30 +162,29 @@ ABORT
-                           ->  Hash                                                                
-                                 ->  Broadcast Motion 3:3  (slice2; segments: 3)                   
-                                       ->  Hash Join                                               
-                                             Hash Cond: (tab2.a = tab1.a)                          
-                                             ->  Seq Scan on tab2                                  
-                                             ->  Hash                                              
-                                                   ->  Broadcast Motion 3:3  (slice1; segments: 3) 
-                                                         ->  Seq Scan on tab1                      
- Optimizer: Pivotal Optimizer (GPORCA)                                                             
-(17 rows)
+ QUERY PLAN                                                                                  
+---------------------------------------------------------------------------------------------
+ Update                                                                                      
+   ->  Result                                                                                
+         ->  Redistribute Motion 3:3  (slice3; segments: 3)                                  
+               Hash Key: tab1.b                                                              
+               ->  Hash Join                                                                 
+                     Hash Cond: (tab3.a = tab1.b)                                            
+                     ->  Seq Scan on tab3                                                    
+                     ->  Hash                                                                
+                           ->  Broadcast Motion 3:3  (slice2; segments: 3)                   
+                                 ->  Hash Join                                               
+                                       Hash Cond: (tab2.a = tab1.a)                          
+                                       ->  Seq Scan on tab2                                  
+                                       ->  Hash                                              
+                                             ->  Broadcast Motion 3:3  (slice1; segments: 3) 
+                                                   ->  Seq Scan on tab1                      
+ Optimizer: Pivotal Optimizer (GPORCA)                                                       
+(16 rows)
 begin;
 BEGIN
 update tab1 set a = 999 from tab2, tab3 where tab1.a = tab2.a and tab1.b = tab3.a;
-ERROR:  distribution key of the tuple (0, 1) doesn't belong to current segment (actually from seg0) (nodeModifyTable.c:602)  (seg1 127.0.1.1:6003 pid=78344) (nodeModifyTable.c:602)
+UPDATE 1
 abort;
```

Comment:
Why is it still necessary to have an abortion if it’s successful?

Reply:
we perform an abort not because the query fails, but because we don't want to save the result.